journal/src/main/java/io/camunda/zeebe/journal/file/SegmentedJournalReader.java

```diff
@@ -64,17 +64,11 @@ public JournalRecord next() {
   }
 
   private JournalRecord unsafeNext() throws NoSuchElementException {
-    if (!currentReader.hasNext()) {
-      final JournalSegment nextSegment = journal.getNextSegment(currentSegment.index());
-      if (nextSegment != null && nextSegment.index() == getNextIndex()) {
-        replaceCurrentSegment(nextSegment);
-        return currentReader.next();
-      } else {
-        throw new NoSuchElementException();
-      }
-    } else {
-      return currentReader.next();
```

Comment:
:question: I'm confused - this was removed, but it looks like that was also already in the `unsafeHasNext()`? What was the duplication for?

Reply:
This was just a duplicate of unsafeHasNext(). Realized it when I had to edit both code.

Reply:
I don't see any need for this duplication unless I'm overlooking something.